src/rpc/core_rpc_server.cpp

```diff
@@ -2606,6 +2622,7 @@ namespace cryptonote
   bool core_rpc_server::on_get_coinbase_tx_sum(const COMMAND_RPC_GET_COINBASE_TX_SUM::request& req, COMMAND_RPC_GET_COINBASE_TX_SUM::response& res, epee::json_rpc::error& error_resp, const connection_context *ctx)
   {
     RPC_TRACKER(get_coinbase_tx_sum);
+    db_rtxn_guard rtxn_guard(&m_core.get_blockchain_storage().get_db());
```

Comment:
This could get the blockchain db locked for an awfully long time, which might hold up all other things quite a bit, but at least the sum would be correct to the piconero :)

Reply:
It's using a read-only LMDB transaction to ensure consistent reads of the db, which doesn't lock out writers. From the [LMDB docs](http://www.lmdb.tech/doc/starting.html):

> as long as a transaction is open, a consistent view of the database is kept alive, which requires storage. A read-only transaction that no longer requires this consistent view should be terminated (committed or aborted) when the view is no longer needed (but see below for an optimization).

> There can be multiple simultaneously active read-only transactions but only one that can write. Once a single read-write transaction is opened, all further attempts to begin one will block until the first one is committed or aborted. This has no effect on read-only transactions, however, and they may continue to be opened at any time.

I also tested behavior in this PR generally by using `usleep` + a local test env so I can mine blocks quickly. Can test behavior like this for example:

```
bool core_rpc_server::on_get_coinbase_tx_sum(const COMMAND_RPC_GET_COINBASE_TX_SUM::request& req, COMMAND_RPC_GET_COINBASE_TX_SUM::response& res, epee::json_rpc::error& error_resp, const connection_context *ctx)
  {
    RPC_TRACKER(get_coinbase_tx_sum);
    db_rtxn_guard rtxn_guard(&m_core.get_blockchain_storage().get_db());
    const uint64_t bc_height = m_core.get_current_blockchain_height();

    printf("Height before sleep: %lu\n", bc_height);
    // sleep, then mine some blocks while asleep, then make sure height still reads the same after waking up
    usleep(1000 * 1000 * 30);

    // can even call status (or other RPC calls) from the daemon while asleep and see *those* would return latest height before this next line prints
    printf("Height after sleep: %lu\n", m_core.get_current_blockchain_height());
```

I could even make an RPC call to `on_get_coinbase_sum` again and grab another read txn guard and the daemon proceeds to fall asleep, while the first `on_get_coinbase_sum` was still asleep.

Reply:
Interesting. So experience from other databases does not carry over 1-to-1 to LMDB, as it can do more in parallel as databases can usually do.

And yeah, your test here really settles the issue.

Reply:
Yes, LMDB is rather unique in its capabilities.